### ISSUE TITLE
Fix Timetabl's offline support (not logged in)

### ISSUE THREAD

**Issue by wensenfriendandextra** [debater-coder/timetabl-app#29]
Currently, I can't use [timetabl.app](https://timetabl.app/) (on desktop) when I'm offline.
I'd like offline access on the desktop Timetabl app or at least some message saying I'm offline.

`/iphone_14_pro.glb` doesn't exist offline

**Comment by debater-coder** [comment]
Timetabl does wotk offline. This appears to be a bug with the landing page. I will look imto it

**Comment by debater-coder** [comment]
Also please note that offline support only works after opening the app once.

**Comment by debater-coder** [comment]
Your stack trace indicates that this bug occurs on the landing page. If I am correct, this bug only occurs when you are not logged in. This occurs due to the page not being able to load a 3D model. I will add some error handling code to handle this case.

When you are logged in, the offline supports kicks in as usual (at least for me):
![image](https://user-images.githubusercontent.com/52619668/205903327-a7c3de12-e706-4988-8f74-d501251d5a1c.png)

**Comment by debater-coder** [comment]
This issue happens on the landing page when the component attempts to load `/iphone_14_pro.glb`, but fails due to the request missing the service worker cache. This could be fixed by adding this file to the cache by default, but it is not really worth caching such a large file (and wasting the user's data and storage) since it is primarily for decorative purposes.

When this fails to load, it causes the React component to go into 'error' state due to an uncaught error occurring in it. Just like in vanilla JS, this error 'bubbles up' to the parent scope (the parent component) and continues to bubble up, terminating everything in its path. This is actually desired behavior as it allows us to have fine grained control over errors and how to handle them to prevent undefined behavior.

This handling is the equivalent of the `try...catch` statement in vanilla JS, and in React they are done using [Error Boundaries](https://reactjs.org/docs/error-boundaries.html). For a while now I have been meaning to place Error Boundaries between distinct sections of the app, and this is the perfect chance to finally implement this.

> TLDR; The solution is to place an [Error Boundary](https://reactjs.org/docs/error-boundaries.html) around the `HeroExperience` component, and it may also be a good time to place Error Boundaries around different routes and around other error-prone components.

I can fix this myself, but if you want I could assign it to you, since it would help increase your familiarity with the codebase. If you don't feel ready, it's no big deal since I can do this myself.

**Comment by wensenfriendandextra** [comment]
REMOVE THE IPHONE THEN >:(((((((((((((((((((((

also iphone gives mme some performance issues
maybe its just my *really bad* computer tho

**Comment by debater-coder** [comment]
The iPhone does look nice though. There shouldn't be any issues with performance, so if you are having issues please send me a screencast. I will add an error boundary, and when I have time I will add performance optimizations.
